src/node/build/index.ts

```diff
@@ -564,9 +570,13 @@ export async function build(
             (plugin) => plugin.name !== 'vite:emit'
           ),
           // vite:emit
-          createEmitPlugin(emitAssets, async (assets) => {
-            indexHtml = emitIndex ? await renderIndex(assets) : ''
-            result = { build, assets, html: indexHtml }
+          createEmitPlugin(emitAssets, async (assets, name) => {
+            // #1071 ignore bundles from rollup-plugin-worker-loader
+            if (name !== outputOptions.name) return
```

Comment:
`rollup-plugin-worker-loader` generates a bundle using the same options of the bundle it was plugged into, which results in `vite:emit` being used by it. Since the worker bundle is generated and not written, we can safely bail out here when the output name is different.